### PR TITLE
add TOPIC parsing

### PR DESCRIPTION
**Pull request by Yay295**

### CODE REVIEW

**Comment by numberoverzero** [comment]
There's a similar (but not sufficient) test in `test_unpack.py::test_channel_message_commands` that should be expanded/modified to cover TOPIC.

Want me to take care of that, or are you up for it?  You've tackled quite a bit lately :)

**Comment by Yay295** [comment]
I think I'd just need to add "TOPIC" to cmd? I don't know where 'nick' comes from in `message = command + " nick #ch :m"` though, for any of those commands.

File: bottom/unpack.py [hunk 1]
```diff
@@ -160,7 +160,8 @@
     ("485", "ERR_UNIQOPPRIVSNEEDED"),
     ("491", "ERR_NOOPERHOST"),
     ("501", "ERR_UMODEUNKNOWNFLAG"),
-    ("502", "ERR_USERSDONTMATCH")
+    ("502", "ERR_USERSDONTMATCH"),
+    ("TOPIC", "TOPIC")
```
Comment (numberoverzero) [inline bottom/unpack.py]: Just remembered why this line isn't needed: synonym lookup falls back to the normalized input value:

```python
def synonym(command: str) -> str:
    command = command.upper()
    return _2812_synonyms.get(command, command)
```

(yep, more cruft to burn away)

**Comment by numberoverzero** [comment]
Yep, that should do it.  TOPIC has an optional message while the others don't, but that's easy enough to add elsewhere.

**Comment by Yay295** [comment]
How about that?

**Comment by numberoverzero** [comment]
ah, that's my bad.  I skimmed the definition of topic and missed that channel is always first, followed by optional message.

you can just revert the test change and I'll add that in.

**Comment by numberoverzero** [comment]
Thanks! 🍰

**Pull request merged**